Utilities/Classes/JSONDataDecoder.swift

```diff
@@ -0,0 +1,141 @@
+     
+     - Returns: Object of data type T containing decoded data
+     */
+    
+    public static func fromData<T: Decodable>(data: Data, keyCodingStrategy: JSONDecoder.KeyDecodingStrategy = .convertFromSnakeCase) throws -> T {
+        
+        let decoder = JSONDecoder()
+        decoder.keyDecodingStrategy = keyCodingStrategy
+        
+        do {
+            let object = try decoder.decode(T.self, from: data)
+            return object
+        } catch {
+            let logger = OSLog(subsystem: subsystem, category:  "error");
+            os_log(msgDataNotDecoded, log: logger, type: .debug)
+            throw JSONDataDecoder.DecoderError.dataCouldNotBeDecoded
+        }
+    }
+    
+    
+    /**
+     Creates an object of data type T, which contains data decoded from conttents of file url.
+     
+     - Parameter fileUrl: URL object, whoose data will be decoded
+     - Parameter keyCodingStrategy: Key decoding strategy, e.g. convertFromSnakeCase or useDefaultKeys (optional)
+     
+     - Returns: Object of data type T containing decoded data
+     */
+    
+    public static func fromURL<T: Decodable>(fileUrl: URL, keyCodingStrategy: JSONDecoder.KeyDecodingStrategy = .convertFromSnakeCase) throws -> T {
+        
+        #if DEBUG
+            let urlComp = URLComponents(url: fileUrl, resolvingAgainstBaseURL: false)
+            assert(urlComp?.scheme == "file")
+        #endif
+
+        var data:Data
+        do {
+            data = try Data(contentsOf: fileUrl)
+        } catch {
+            let logger = OSLog(subsystem: subsystem, category:  "error");
+            os_log(msgContentsOFUrlNotLoaded ,log: logger, type: .debug, fileUrl.absoluteString)
+            
+            throw JSONDataDecoder.DecoderError.contentsOfUrlCouldNotBeLoaded
+        }
+        
+        let decoder = JSONDecoder()
+        decoder.keyDecodingStrategy = keyCodingStrategy
+        
+        do {
+            let object = try decoder.decode(T.self, from: data)
+            return object
+        } catch {
+            let logger = OSLog(subsystem: subsystem, category:  "error");
+            os_log(msgDataNotDecoded, log: logger, type: .debug)
+            throw JSONDataDecoder.DecoderError.dataCouldNotBeDecoded
+        }
+    }
+    
+    
+    /**
+     Creates an object of data type T, which contains data decoded from json file.
+     
+     - Parameter fileName: Name of the json file
+     - Parameter fileExtension: Extension of json file
+     - Parameter keyCodingStrategy: Key decoding strategy, e.g. convertFromSnakeCase or useDefaultKeys  (optional)
+     
+     - Returns: Object of data type T containing decoded data
+     */
+    public static func fromLocalFile<T: Decodable>(fileName: StaticString, fileExtension: String? = "json", keyCodingStrategy: JSONDecoder.KeyDecodingStrategy = .convertFromSnakeCase) throws -> T {
+        
+        if let fileUrl = Bundle.main.url(forResource: "\(fileName)", withExtension: fileExtension) {
```

Comment:
Um zu tiefe Schachteln zu vermeiden, bietet es sich an, an dieser Stelle ein 'guard'-Statement zu verwenden. Also in etwa so: 
guard let fileUrl = Bundle.main.url(...) else {
   <code der ausgeführt wird, wenn Datei nicht gefunden...>
   return 
}

Normaler Programmfluss...